fsd_utils/authentication/decorators.py

```diff
@@ -50,30 +62,42 @@ def _check_access_token(auto_redirect=True):
         return False
 
 
-def login_required(f):
+def login_required(f=None, roles_required: List[str] = None):
     """
-    Execute function if request contains valid JWT
-    and pass account auth params to route as attributes
-    on the flask request global 'g' object:
-        - g.account_id: (str) users account id
-        - g.is_authenticated: (bool) authentication status
-        - g.user - this holds a User object and associated attributes
-        - g.logout_url: (str) the service sign-out url
-    If no valid auth JWT found then redirect to
-    service config invalid login route
+     Execute function if request contains valid JWT
+     and pass account auth params to route as attributes
+     on the flask request global 'g' object:
+         - g.account_id: (str) users account id
+         - g.is_authenticated: (bool) authentication status
+         - g.user - this holds a User object and associated attributes
+         - g.logout_url: (str) the service sign-out url
+     If no valid auth JWT found then redirect to
+     service config invalid login route
+    :param f:
+    :param roles_required: (List(str), optional) a list of the
+            roles required to access the decorated route
+    :return:
     """
+    if f is None:
+        return lambda f: login_required(f=f, roles_required=roles_required)
```

Comment:
what is the purpose of f here? will the login required func run in a cycle if f = None?

Reply:
This is so that the `@login_required` decorator can work with or without arguments, eg `@login_required` or `@login_required(roles_required=ASSESSOR)`